src/servers/FtpServer.cc

```diff
@@ -241,21 +241,24 @@ Ftp::Server::noteBodyConsumerAborted(BodyPipe::Pointer ptr)
 void
 Ftp::Server::AcceptCtrlConnection(const CommAcceptCbParams &params)
 {
-    MasterXaction::Pointer xact = params.xaction;
-    AnyP::PortCfgPointer s = xact->squidPort;
+    Assure(params.port);
 
     // NP: it is possible the port was reconfigured when the call or accept() was queued.
 
     if (params.flag != Comm::OK) {
         // Its possible the call was still queued when the client disconnected
-        debugs(33, 2, s->listenConn << ": FTP accept failure: " << xstrerr(params.xerrno));
+        debugs(33, 2, params.port->listenConn << ": FTP accept failure: " << xstrerr(params.xerrno));
         return;
     }
 
     debugs(33, 4, params.conn << ": accepted");
     fd_note(params.conn->fd, "client ftp connect");
 
+    auto xact = MasterXaction::MakePortful(params.port);
```

Comment:
I may be missing something, but it looks like this and several very similar places can use `const` because the `xact` pointer is not modified and the Server class constructor accepts a constant pointer. Please check and, if possible, add `const` here and in similar such places.

Reply:
I agree that we can use const in this context. Making the similar fix in other contexts required making NewServer() argument constant. Done at e6ae845.